central/notifiers/cscc/cscc.go

```diff
@@ -50,102 +54,99 @@ func init() {
 	})
 }
 
+type config struct {
+	ServiceAccount string `json:"serviceAccount"`
+	SourceID       string `json:"sourceID"`
+}
+
 // The Cloud SCC notifier plugin integrates with Google's Cloud Security Command Center.
 type cscc struct {
 	// The Service Account is a Google JSON service account key.
 	// The GCP Organization ID is a numeric identifier for the Google Cloud Platform
 	// organization. It is required so that we can tag findings to the right org.
-	client client.Config
+	client *securitycenter.Client
 	config *config
 	*storage.Notifier
 }
 
-type config struct {
-	ServiceAccount string `json:"serviceAccount"`
-	SourceID       string `json:"sourceID"`
-}
+func newCSCC(protoNotifier *storage.Notifier, cryptoCodec cryptocodec.CryptoCodec, cryptoKey string) (*cscc, error) {
+	conf := protoNotifier.GetCscc()
+	if err := Validate(conf, !env.EncNotifierCreds.BooleanSetting()); err != nil {
+		return nil, err
+	}
 
-// Validate Cscc notifier
-func Validate(cscc *storage.CSCC, validateSecret bool) error {
-	if validateSecret && cscc.ServiceAccount == "" {
-		return errors.New("serviceAccount must be defined in the Cloud SCC Configuration")
+	decCreds := conf.ServiceAccount
+	var err error
+	if env.EncNotifierCreds.BooleanSetting() {
+		if protoNotifier.GetNotifierSecret() == "" {
+			return nil, errors.Errorf("encrypted notifier credentials for notifier '%s' empty", protoNotifier.GetName())
+		}
+		decCreds, err = cryptoCodec.Decrypt(cryptoKey, protoNotifier.GetNotifierSecret())
+		if err != nil {
+			return nil, errors.Errorf("Error decrypting notifier secret for notifier '%s'", protoNotifier.GetName())
+		}
 	}
-	if cscc.SourceId == "" {
-		return errors.New("sourceID must be defined in the Cloud SCC Configuration")
+
+	cfg, err := google.CredentialsFromJSON(context.Background(), []byte(decCreds), secretmanager.DefaultAuthScopes()...)
```

Comment:
Why use the secretmanager scope here? I imagine it's most likely the same as scc, but in principle every client has its scope that may also change in the future. I would suggest to use https://github.com/googleapis/google-api-go-client/blob/main/securitycenter/v1/securitycenter-gen.go#L99 instead.

Reply:
Hmm, the reason being that it's the documented way to create a new client ([see here](https://pkg.go.dev/cloud.google.com/go#hdr-Authentication_and_Authorization)), but I'm fine changing it to use the different one.